pgmax/groups/enumeration.py

```diff
@@ -231,36 +224,37 @@ def __post_init__(self):
             )
 
         if log_potential_matrix.ndim == 3 and log_potential_matrix.shape[0] != len(
-            self.variable_names_for_factors
+            self.variables_for_factors
         ):
             raise ValueError(
-                f"Expected log_potential_matrix for {len(self.variable_names_for_factors)} factors. "
+                f"Expected log_potential_matrix for {len(self.variables_for_factors)} factors. "
                 f"Got log_potential_matrix for {log_potential_matrix.shape[0]} factors."
             )
 
-        for fac_list in self.variable_names_for_factors:
-            if len(fac_list) != 2:
+        import time
+
+        start = time.time()
+        log_potential_shape = log_potential_matrix.shape[-2:]
+        for variables_for_factor in self.variables_for_factors:
+            if len(variables_for_factor) != 2:
                 raise ValueError(
                     "All pairwise factors should connect to exactly 2 variables. Got a factor connecting to"
-                    f" {len(fac_list)} variables ({fac_list})."
+                    f" {len(variables_for_factor)} variables ({variables_for_factor})."
                 )
 
-            # Note: num_states0 = self.variable_group[fac_list[0]] is 2x slower
-            num_states0 = self.variable_group._names_to_variables[
-                fac_list[0]
-            ].num_states
-            num_states1 = self.variable_group._names_to_variables[
-                fac_list[1]
-            ].num_states
-
-            if not log_potential_matrix.shape[-2:] == (num_states0, num_states1):
+            factor_num_configs = (
+                variables_for_factor[0][1],
+                variables_for_factor[1][1],
+            )
+            if log_potential_shape != factor_num_configs:
                 raise ValueError(
-                    f"The specified pairwise factor {fac_list} (with "
-                    f"{(self.variable_group[fac_list[0]].num_states, self.variable_group[fac_list[1]].num_states)} "
+                    f"The specified pairwise factor {variables_for_factor} (with {factor_num_configs}"
                     f"configurations) does not match the specified log_potential_matrix "
-                    f"(with {log_potential_matrix.shape[-2:]} configurations)."
+                    f"(with {log_potential_shape} configurations)."
                 )
-        object.__setattr__(self, "log_potential_matrix", log_potential_matrix)
+            object.__setattr__(self, "log_potential_matrix", log_potential_matrix)
```

Comment:
Wrong indentation, you are setting this many times

Reply:
thanks :)